### PR TITLE
WIP - Intune - Add code example

### PR DESCRIPTION
**Pull request by burkeholland**

### CODE REVIEW

**Comment by herohua** [comment]
### :x: Validation status: errors


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-conceptual/index.yml](https://github.com/burkeholland/azure-docs-sdk-node/blob/patch-15/docs-ref-conceptual/index.yml) | :x:Error | | [Details](#user-content-docs-ref-conceptual%2Findex.yml)

<a name="docs-ref-conceptual%2Findex.yml"></a>
### [docs-ref-conceptual/index.yml](https://github.com/burkeholland/azure-docs-sdk-node/blob/patch-15/docs-ref-conceptual/index.yml)
 - **[Error]** File docs-ref-conceptual/index.yml with type LandingData doesn&#39;t have asset id. Please set it.

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C7%5C13%5Cb6261451-6613-d51e-95b5-8b5dc96f3a2c%5CPullRequest%5C201707131952467546-80%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by TomArcher** [comment]
@burkeholland Don't worry about the build error. We're looking into it. Just #sign-off whenever you're ready, and I'll merge :)

File: docs-ref-services/intune.md [hunk 1]
```diff
@@ -24,17 +24,33 @@ Azure Intune is a cloud-based enterprise mobility management (EMM) service that
 
 Intune integrates closely with Azure Active Directory (Azure AD) for identity and access control, and Azure Information Protection for data protection.
 
-## Install the modules with npm
+## Management Package
+
+### Install npm modules
 
 Use npm to install the Azure Intune modules for Node.js
```
Comment (johnpapa) [inline docs-ref-services/intune.md]: modules --> module

File: docs-ref-services/intune.md [hunk 2]
```diff
@@ -24,17 +24,33 @@ Azure Intune is a cloud-based enterprise mobility management (EMM) service that
...
+
+Authenticate, create a client and retrieve a location by host name.
+
+```javascript
+const msRestAzure = require('ms-rest-azure');
+const IntuneResourceManagementClient = require('azure-arm-intune');
+
+msRestAzure.interactiveLogin().then(credentials => {
+  const client = new IntuneResourceManagementClient(credentials);
+  client.getLocationByHostName(null).then((error, result) => {
```
Comment (johnpapa) [inline docs-ref-services/intune.md]: is that the right call signature? isnt `error` part of the `catch` ?

Comment (johnpapa) [inline docs-ref-services/intune.md]: https://github.com/Azure/azure-sdk-for-node/blob/e90415ddd004e32a7ceff548a3d47d34866ec09c/lib/services/intune/lib/intuneResourceManagementClient.d.ts#L125

I think that is it is a Promise with a single result. see the link above

File: docs-ref-services/intune.md (hunk 1 above)
Comment (burkeholland) [inline docs-ref-services/intune.md]: Where? In the header or in the text or both?

**Comment by burkeholland** [comment]
@johnpapa Changes are in. Ready to merge?

**Comment by herohua** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-services/intune.md](https://github.com/burkeholland/azure-docs-sdk-node/blob/patch-15/docs-ref-services/intune.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C7%5C13%5Cb6261451-6613-d51e-95b5-8b5dc96f3a2c%5CPullRequest%5C201707132033408605-80%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by herohua** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-services/intune.md](https://github.com/burkeholland/azure-docs-sdk-node/blob/patch-15/docs-ref-services/intune.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C7%5C13%5Cb6261451-6613-d51e-95b5-8b5dc96f3a2c%5CPullRequest%5C201707132035528807-80%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs-ref-services/intune.md [hunk 3]
```diff
@@ -24,17 +24,33 @@ Azure Intune is a cloud-based enterprise mobility management (EMM) service that
...
+
+Authenticate, create a client and retrieve a location by host name.
+
+```javascript
+const msRestAzure = require('ms-rest-azure');
+const IntuneResourceManagementClient = require('azure-arm-intune');
+
+msRestAzure.interactiveLogin().then(credentials => {
+  const client = new IntuneResourceManagementClient(credentials);
+  client.getLocationByHostName(null).then(result => {
```
Comment (johnpapa) [inline docs-ref-services/intune.md]: host name is null. shouldnt this be the name we search for?

Comment (burkeholland) [inline docs-ref-services/intune.md]: That's a goooooood question. I _think_ this works with `null` in place of an options object (looking at the code), but I need to check with someone who knows. I don't have permissions to read over the Microsoft InTune scope, so I can't actually get that far.

Comment (johnpapa) [inline docs-ref-services/intune.md]: OK. Let's hold off on this one til you get an answer

File: docs-ref-services/intune.md [hunk 4]
```diff
@@ -24,17 +24,33 @@ Azure Intune is a cloud-based enterprise mobility management (EMM) service that
 
 Intune integrates closely with Azure Active Directory (Azure AD) for identity and access control, and Azure Information Protection for data protection.
 
-## Install the modules with npm
+## Management Package
 
-Use npm to install the Azure Intune modules for Node.js
+### Install npm module
+
+Use npm to install the Azure Intune module for Node.js
```
Comment (johnpapa) [inline docs-ref-services/intune.md]: > Install the Azure Intune npm module

File: docs-ref-services/intune.md [hunk 5]
```diff
@@ -24,17 +24,33 @@ Azure Intune is a cloud-based enterprise mobility management (EMM) service that
...
+  const client = new IntuneResourceManagementClient(credentials);
+  client.getLocationByHostName(null).then(result => {
+    console.log(result);
+  });
+});
+```
+
 
-##Samples
+### Samples
```
Comment (johnpapa) [inline docs-ref-services/intune.md]: 2 ##'s only

**Comment by herohua** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-services/intune.md](https://github.com/burkeholland/azure-docs-sdk-node/blob/patch-15/docs-ref-services/intune.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C7%5C18%5Cb6261451-6613-d51e-95b5-8b5dc96f3a2c%5CPullRequest%5C201707180103397501-80%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by herohua** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-services/intune.md](https://github.com/burkeholland/azure-docs-sdk-node/blob/patch-15/docs-ref-services/intune.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C7%5C18%5Cb6261451-6613-d51e-95b5-8b5dc96f3a2c%5CPullRequest%5C201707180424385673-80%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by johnpapa** [comment]
@burkeholland  can you follow up on this one ?

**Comment by johnpapa** [comment]
@burkeholland  status update?

**Comment by burkeholland** [comment]
Intune is dead. This API is being removed from the sidebar. Closing this PR unceremoniously.

**Pull request closed without merging**